tracktolib/pg_sync.py

```
@@ -104,21 +104,33 @@ def insert_one(
+        engine.commit()
+    else:
+        _ = engine.execute(query, _data[0])
         if _is_returning:
-            resp = cur.fetchone()
-        else:
-            resp = None
-    engine.commit()
+            row = engine.fetchone()
+            if row is not None:
+                if engine.description is None:
+                    raise RuntimeError("engine.description is None — cannot build dict from row.")
```

Ruff (0.13.1)

Avoid specifying long messages outside the exception class

(TRY003)